include/matx/core/tensor_impl.h

```diff
@@ -110,13 +113,15 @@ class tensor_impl_t {
 
       swap(lhs.ldata_, rhs.ldata_);
       swap(lhs.desc_, rhs.desc_);
+      swap(lhs.stf_ldata_, rhs.stf_ldata_);
     }
 
     /**
      * Constructor for a rank-0 tensor (scalar).
      */
     tensor_impl_t() {
-
+      auto ldptr = new std::optional<stf_logicaldata_type>();
```

Comment:
This won't compile anymore since we don't allow std:: types on the device. It might work with cuda::std::optional, but we don't use that anywhere currently.